src/Extractor/SnowflakeOdbcConnection.php

```diff
@@ -4,12 +4,41 @@
 
 namespace Keboola\DbExtractor\Extractor;
 
+use Keboola\DbExtractor\Adapter\Exception\OdbcException;
 use Keboola\DbExtractor\Adapter\ODBC\OdbcConnection;
+use Throwable;
 
 class SnowflakeOdbcConnection extends OdbcConnection
 {
     use QuoteTrait;
 
+    protected function connect(): void
+    {
+        $this->logger->info(sprintf('Creating ODBC connection to "%s".', $this->dsn));
+        ini_set('odbc.default_cursortype', (string) $this->odbcCursorType);
+        try {
+            /** @var resource|false $connection */
+            $connection = @odbc_connect($this->dsn, $this->user, $this->password, $this->odbcCursorMode);
+        } catch (Throwable $e) {
+            throw new OdbcException($e->getMessage(), $e->getCode(), $e);
+        }
+
+        // "odbc_connect" can generate warning, if "set_error_handler" is not set, so we are checking it manually
+        if ($connection === false) {
+            if (strpos(odbc_errormsg(), 'msg=')) {
```

Comment:
tomu nerozumim, proč to není tady https://github.com/keboola/db-extractor-adapter/blob/25b9cb7c59e7a7ae65a68df7dc7e6f5e3ce868f4/src/ODBC/OdbcConnection.php#L71 ?

Reply:
protože to je specifický pro snflk, ale udělám to ještě teda trochu jinak

Reply:
https://github.com/keboola/db-extractor-adapter/pull/21